### ISSUE TITLE
Event types -- necessary?

### ISSUE THREAD

**Issue by swcurran** [digitalbazaar/cel-spec#7]
A trivial question, but are the event types of create/update/deactivate really needed?  They are pretty implicit -- first, last and all others.  I guess deactivate is a bit tricky, but seems like a lot of overhead for very little.

**Comment by msporny** [comment]
Yes, that's an open question. We had types for logs, events, and operations... and removed the types for logs and events, but kept them for operations. 

Fundamentally, it's a layering question -- do the types of events go at the application layer (in the `data` field)? Or do they go up a level?

One use case we are exploring is the idea that you can have multiple operations per event, where you might create two data objects with different IDs (remember it's the application's state machine that handles how those two objects change over time... or if a third one springs into existence a week from when the second object came into existence). So, you might "create" multiple objects over time... create isn't only the first operation performed... it might also happen during the 50th operation.

Again, this feels like it would add complexity to the spec (and it's not clear we need it).

**Comment by gobengo** [comment]
> do the types of events go at the application layer (in the data field)?

I like them in the data field tagged with a media type unless/until there is a clear reason to lift them up a level.

The events themselves should have a media type. We can reuse the log structure regardless of event type. It's less flexible to lift the types up. Any interop profile is likely going to allowlist the media types anyway, so it doesn't seem like too much flexibility.
